kyuubi-common/src/main/scala/org/apache/kyuubi/events/EventBus.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.events
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.{classTag, ClassTag}
+import scala.util.Try
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.util.ThreadUtils
+
+sealed trait EventBus {
+  def post[T <: KyuubiEvent](event: T): Unit
+
+  def register[T <: KyuubiEvent: ClassTag](et: EventHandler[T]): EventBus
+
+  def registerAsync[T <: KyuubiEvent: ClassTag](et: EventHandler[T]): EventBus
+}
+
+object EventBus extends Logging {
+  private val defaultEventBus = EventBusLive()
+
+  def apply(): EventBus = EventBusLive()
+
+  // Exposed api
+  def post[T <: KyuubiEvent](event: T): Unit = defaultEventBus.post[T](event)
+
+  def register[T <: KyuubiEvent: ClassTag](et: EventHandler[T]): EventBus =
+    defaultEventBus.register[T](et)
+
+  def registerAsync[T <: KyuubiEvent: ClassTag](et: EventHandler[T]): EventBus =
+    defaultEventBus.registerAsync[T](et)
+
+  private case class EventBusLive() extends EventBus {
+    private[this] val eventHandlerRegistry = new Registry
+    private[this] val asyncEventHandlerRegistry = new Registry
+    implicit private[this] val ec: ExecutionContext =
+      ExecutionContext.fromExecutor(
+        ThreadUtils.newDaemonQueuedThreadPool(10, 10, 60000L, "event-bus-exec-pool"))
```

Comment:
The queue size seems to be too small, and how do we handle subsequent events when the queue is full?

Reply:
For asynchronous processing is relatively simple to implement, the size of the thread pool is configured by kyuubi Config or changed to a ForkJoin thread pool?